fluss-common/src/main/java/com/alibaba/fluss/metadata/DatabaseDescriptor.java

```diff
@@ -0,0 +1,145 @@
+
+import javax.annotation.Nullable;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Represents the metadata of a database in Fluss.
+ *
+ * <p>It contains all characteristics that can be expressed in a SQL {@code CREATE Database}
+ * statement, such as schema, primary keys, partition keys, bucket keys, and options.
+ *
+ * @since 0.6
+ */
+public class DatabaseDescriptor {
+    private final Map<String, String> properties;
+    private final String comment;
+
+    private DatabaseDescriptor(Map<String, String> properties, @Nullable String comment) {
+        this.properties = properties;
+        this.comment = comment;
+    }
+
+    public Map<String, String> getProperties() {
+        return properties;
+    }
+
+    public Optional<String> getComment() {
+        return Optional.ofNullable(comment);
+    }
+
+    /**
+     * Serialize the table descriptor to a JSON byte array.
+     *
+     * @see DatabaseDescriptorJsonSerde
+     */
+    public byte[] toJsonBytes() {
+        return JsonSerdeUtils.writeValueAsBytes(this, DatabaseDescriptorJsonSerde.INSTANCE);
+    }
+
+    /**
+     * Deserialize from JSON byte array to an instance of {@link DatabaseDescriptor}.
+     *
+     * @see DatabaseDescriptor
+     */
+    public static DatabaseDescriptor fromJsonBytes(byte[] json) {
+        return JsonSerdeUtils.readValue(json, DatabaseDescriptorJsonSerde.INSTANCE);
+    }
+
+    /** Creates a builder for building database descriptor. */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    // ---------------------------------------------------------------------------------------------
+
+    /** Builder for {@link TableDescriptor}. */
+    @PublicEvolving
+    public static class Builder {
+
+        private final Map<String, String> properties;
+        private @Nullable String comment;
+
+        protected Builder() {
+            this.properties = new HashMap<>();
+        }
+
+        protected Builder(TableDescriptor descriptor) {
+            this.properties = new HashMap<>(descriptor.getProperties());
+            this.comment = descriptor.getComment().orElse(null);
+        }
+
+        /**
+         * Sets table property on the table.
+         *
+         * <p>Table properties are controlled by Fluss and will change the behavior of the table.
+         */
+        public <T> Builder property(ConfigOption<T> configOption, T value) {
+            Preconditions.checkNotNull(configOption, "Config option must not be null.");
+            Preconditions.checkNotNull(value, "Value must not be null.");
+            properties.put(
+                    configOption.key(), ConfigurationUtils.convertValue(value, String.class));
+            return this;
+        }
+
+        /**
+         * Sets table property on the table.
+         *
+         * <p>Table properties are controlled by Fluss and will change the behavior of the table.
```

Comment:
```suggestion
         * <p>Database properties are controlled by Fluss
```
No change of the behavior of the database...